connlimit.go

```diff
@@ -182,11 +192,23 @@ func (l *Limiter) SetConfig(c Config) {
 // their own calls if they need to continue limiting the number of concurrent
 // hijacked connections.
 func (l *Limiter) HTTPConnStateFunc() func(net.Conn, http.ConnState) {
+	message := "Your IP is issuing too many concurrent connections, please rate limit your calls\n"
+	tooManyRequestsResponse := []byte(fmt.Sprintf("HTTP/1.1 429 Too Many Requests\r\n"+
+		"Content-Type: text/plain\r\n"+
+		"Content-Length: %d\r\n"+
+		"Connection: close\r\n\r\n%s", len(message), message))
 	return func(conn net.Conn, state http.ConnState) {
 		switch state {
 		case http.StateNew:
 			_, err := l.Accept(conn)
 			if err != nil {
+				if err == ErrPerClientIPLimitReached {
+					// We don't care about slow players
+					if l.RWDealineMaxDelay > 0 {
+						conn.SetDeadline(time.Now().Add(l.RWDealineMaxDelay))
+					}
+					conn.Write(tooManyRequestsResponse)
```

Comment:
I just realized why reading wasn't a data race, and it might cause problems for this approach.

[`ConnState(conn, http.StateNew)`](https://github.com/golang/go/blob/release-branch.go1.14/src/net/http/server.go#L2901-L2932) is called *serially* after the HTTP Server's `listener.Accept()`.

That means even a very short `RWDealineMaxDelay` (need to fix the typo in Dealine) blocks *all other new connections.*

We could attempt to do the Write and Close in a goroutine, but I have no idea what impact that might have on the HTTP server.

Reply:
@schmichaelI I would think it has low impact compared to having just conn.Close(), furthermore it happens only when rate limited. (we send less than 200 bytes, will probably take as much net usage as just closing. 
I don't think the go routine is worth the cost.
But you choose of course